frontend/scripts/componentSchemas/schemaUtils.ts

```diff
@@ -88,37 +160,27 @@ export const verifySchema = (schema: any, expectedProperties: string[]) => {
     console.log(`Missing properties: ${missingProperties.join(', ')}`);
     return false;
   }
-
   return true;
 };
 
 /**
- * Expands all refs in properties in a schema
- * @param properties The properties to expand
- * @param layoutSchema The full layout schema
- * @returns The expanded properties
+ * Ensures that a schema with enum values has correct type (string or number)
+ * @param schema The schema to ensure type for
  */
-export const expandRefsInProperties = (properties: any, layoutSchema: any) => {
-  const expandedProperties = { ...properties };
-  for (const property in properties) {
-    if (expandedProperties[property].$ref && expandedProperties[property].$ref.startsWith('#/')) {
-      expandedProperties[property] = expandRef(expandedProperties[property].$ref, layoutSchema);
-    }
-    if (expandedProperties[property].items?.$ref) {
-      expandedProperties[property].items = expandRef(
-        expandedProperties[property].items.$ref,
-        layoutSchema,
-      );
-    }
-    if (expandedProperties[property].allOf) {
-      expandedProperties[property] = expandAllOf(expandedProperties[property], layoutSchema);
+export const ensureTypeWithEnums = (schema: any) => {
```

Comment:
Maybe we could look at the possibility to split this more too? 🤔  😄

Reply:
Something like this?
```ts
export const ensureTypeWithEnums = (schema: any) => {
  if (schema.enum && schema.enum.length > 0) {
    setTypeFromEnum(schema);
  } else if (schema.items?.enum && schema.items.enum.length > 0) {
    setTypeFromEnum(schema.items);
  }
};

const setTypeFromEnum = (schema: any) => {
  const inferredType = inferTypeFromEnum(schema.enum);
  if (inferredType) {
    schema.type = inferredType;
  }
};

const inferTypeFromEnum = (enumValues: string[] | number[]) => {
  const firstEnumValue = enumValues[0];
  if (typeof firstEnumValue === 'string') {
    return 'string';
  } else if (typeof firstEnumValue === 'number') {
    return 'number';
  }
  return undefined;
};
```

Reply:
Great! 😄 This looks a lot cleaner! Only the `any` type now 😅 Maybe we can use generics here as well?

Reply:
Fixed with this generic type from `packages/shared` 😊 
```
KeyValuePairs<T = any> {
  [key: string]: T;
}
```